debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -103,15 +107,17 @@ public void execute(ChangeEventSourceContext context) throws InterruptedExceptio
                     continue;
                 }
                 // There is no change in the database
-                if (currentMaxLsn.equals(lastProcessedPosition.getCommitLsn())) {
+                if (currentMaxLsn.equals(lastProcessedPosition.getCommitLsn()) && shouldIncreaseFromLsn) {
```

Comment:
Why is `shouldIncreaseFromLsn` checked here, too?

Reply:
If a transaction was not processed completely and connector was restarted then the unfinished transaction would not be completed until a new transaction was written (increasing maxlsn).